test-app/src/message-board.ts

```diff
@@ -59,95 +59,100 @@ const api = new appsync.GraphqlApi(stack, "Api", {
   },
 });
 
-const getPostResolver = new AppsyncResolver<
-  { postId: string },
-  Post | undefined
->(($context) => {
-  return database.getItem({
-    key: {
-      pk: {
-        S: `Post|${$context.arguments.postId}`,
-      },
-      sk: {
-        S: "Post",
-      },
+const getPost = new AppsyncResolver<{ postId: string }, Post | undefined>(
+  stack,
+  "getPost",
+  {
+    api,
+    typeName: "Query",
+    fieldName: "getPost",
```

Comment:
How does this work with `getField` now. Code first AppSync uses the `add*` methods on the api and just ties the resolver to a single field as the typeName may come from the schema objects.

It also requires adding return and args types to the resolver when binding.

I still think that would all look better being fluent.

```ts
const api = new GqLApi();

api.addResolver("Query", "getPost", { .. props }, () => {
   ....
});


api.addQuery("something", appsync.GrapQLType.string(), { args: {... } }, () => {
   ....
});
```

Reply:
Yeah that's my outstanding task. I was thinking of creating its own class. We could also add methods to our own GraphQLApi class

Reply:
Moved it to `AppsyncField` class